### PR TITLE
fix: CU-2hjq08q - related id in graphql, small refactors and alignments

### PR DESCRIPTION
**Pull request by cyp3rius**
## Ticket

https://github.com/VirtusLab/strapi-plugin-navigation/issues/188

## Summary

What does this PR do/solve? 

1. Provides coverage for `related.id` in GraphQL queries only like:
   ```gql
      related {
        id
        attributes {
          __typename

          ... on Page {
            Title
          }

          ... on WithFlowType {
            Name
          }
      }
    }
   ```
2. Fixes visibility of `draft` status for types which doesn't have publish flow enabled
3. Aligns the render type enums to `FLAT`, `TREE` and `RFR` for both REST and GQL API

## Test Plan

1. Run the app
2. Check your GQL navigation render query for any of content types
3. See that you're now able to get `id` and `attributes` properties
4. Do the same for REST for the same and see no difference in the payload comparing to previous version

### CODE REVIEW

**Comment by cyp3rius** [comment]
Task linked: [CU-2hjq08q Id of related is missing in graphql schema](https://app.clickup.com/t/2hjq08q)

**Review by ltsNotMike — APPROVED** [review]

File: server/utils/constant.ts [hunk 1]
```diff
@@ -14,3 +14,8 @@ export const DEFAULT_NAVIGATION_ITEM = {
   slug: "main-navigation",
   visible: true,
 } as const;
+export enum RENDER_TYPES {
+  FLAT = 'FLAT',
+  TREE = 'TREE',
+  RFR = 'RFR'
+};
```
Comment (CodeVoyager) [inline server/utils/constant.ts:21]: [nitpick] meta of contemporary TS is to move away from TS specific constructs like `enum`, decorators, etc.

File: types/utils.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import { isNumber } from "lodash";
 import { Id } from "strapi-typed";
+import { RENDER_TYPES } from "../server/utils";
 import { Navigation, NavigationItem, NavigationItemType, NestedStructure } from "./contentTypes"
 
 export type NavigationActionsCategories = 'toCreate' | 'toUpdate' | 'toRemove';
-export type RenderType = "flat" | "tree" | "rfr";
+export type RenderType = RENDER_TYPES.FLAT | RENDER_TYPES.TREE | RENDER_TYPES.RFR;
```
Comment (CodeVoyager) [inline types/utils.ts:7]:
```suggestion
export type RenderType = typeof keyof RENDER_TYPES;
```
with `Object`

or with `enum`

```suggestion
export type RenderType = RENDER_TYPES;
```

should give the same result

**Review by CodeVoyager — APPROVED** [review]

File: server/utils/constant.ts (hunk 1 above)
Comment (cyp3rius) [reply server/utils/constant.ts]: @CodeVoyager  changed, if fine please resolve :)

**Pull request merged**